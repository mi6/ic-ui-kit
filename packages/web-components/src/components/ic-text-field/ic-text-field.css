@import "../../global/normalize.css";

:host {
  /**
   * @prop --input-width: Width of the input field
   */
  display: block;
}

:host(.fullwidth) {
  width: 100%;
}

::placeholder {
  color: var(--ic-color-tertiary-text);
  opacity: 1;
}

input,
textarea {
  border: 0;
  border-radius: var(--ic-border-radius);
  background-color: var(--ic-architectural-white);
  line-height: 1.5rem;
  letter-spacing: 0.005rem;
  width: 100%;
  padding-right: var(--ic-space-xs);
  padding-left: var(--ic-space-xs);
}

textarea {
  min-height: var(--ic-space-lg);
  resize: vertical;
  padding-top: 0.375rem;
}

input:focus,
textarea:focus {
  border: 0;
  outline: 0;
}

input:disabled,
textarea:disabled {
  color: var(--ic-architectural-200);
}

input.readonly,
textarea.readonly {
  color: var(--ic-color-primary-text);
}

/* Chrome, Safari, Edge */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type="number"] {
  -moz-appearance: textfield;
}

textarea.no-resize {
  resize: none;
}

.maxlengthtext {
  color: var(--ic-color-secondary-text);
}

.no-left-pad {
  padding-left: 0;
}

::slotted([slot="icon"]) {
  fill: var(--ic-color-tertiary-text);
}

.has-value ::slotted([slot="icon"]) {
  fill: var(--ic-color-primary-text);
}

.charcount {
  margin-right: calc(-1 * var(--ic-space-xxxs));
}

input[type="search"]::-webkit-search-cancel-button,
input[type="search"]::-webkit-search-decoration,
input[type="search"]::-webkit-search-results-button,
input[type="search"]::-webkit-search-results-decoration {
  display: none;
}

input[type="search"].truncate-value {
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@media (forced-colors: active) {
  input.readonly,
  textarea.readonly {
    color: canvastext;
  }

  .has-value ::slotted([slot="icon"]) {
    fill: currentcolor;
  }
}
