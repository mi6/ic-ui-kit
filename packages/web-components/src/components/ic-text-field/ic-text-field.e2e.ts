import { newE2EPage } from "@stencil/core/testing";

describe("ic-text-field", () => {
  it("should trigger max length validation when long value set", async () => {
    const page = await newE2EPage();
    await page.setContent('<ic-text-field label="Test label" max-length="20">');

    await page.waitForChanges();

    const element = await page.find("ic-text-field");
    element.setProperty("value", "a really long test value");

    await page.waitForChanges();

    const input = await page.find("ic-text-field >>> #ic-text-field-input-0");

    const value = await input.getProperty("value");
    expect(value).toBe("a really long test value");

    let icon = await page.find("ic-text-field >>> .icon-error");
    let iconVis = await icon.isVisible();
    expect(iconVis).toBeTruthy();

    const validationText = await page.find(
      "ic-text-field >>> #ic-text-field-input-0-validation-text"
    );
    expect(validationText).toEqualText("Maximum length exceeded");

    const charCountText = await page.find(
      "ic-text-field >>> #ic-text-field-input-0-charcount"
    );
    expect(charCountText).toEqualText("24/20");

    await input.press("Backspace");
    await input.press("Backspace");
    await input.press("Backspace");
    await input.press("Backspace");
    expect(charCountText).toEqualText("20/20");

    expect(validationText).toEqualText("");
    icon = await page.find("ic-text-field >>> .icon-error");
    expect(icon).toBeNull();

    await input.press("a");
    expect(charCountText).toEqualText("21/20");
    expect(validationText).toEqualText("Maximum length exceeded");

    icon = await page.find("ic-text-field >>> .icon-error");
    iconVis = await icon.isVisible();
    expect(iconVis).toBeTruthy();
  });

  it("should trigger min/max value validation when lower/higher value set", async () => {
    const page = await newE2EPage();
    await page.setContent(
      '<ic-text-field label="Test label" min=1 max=4 inputmode="numeric" type="number"'
    );

    await page.waitForChanges();

    const element = await page.find("ic-text-field");
    element.setProperty("value", "5");

    await page.waitForChanges();

    const input = await page.find("ic-text-field >>> #ic-text-field-input-0");

    const value = await input.getProperty("value");
    expect(value).toBe("5");

    const icon = await page.find("ic-text-field >>> .icon-error");
    const iconVis = await icon.isVisible();
    expect(iconVis).toBeTruthy();

    let validationText = await page.find(
      "ic-text-field >>> #ic-text-field-input-0-validation-text"
    );
    expect(validationText).toEqualText("Maximum value of 4 exceeded");

    element.setProperty("value", "4");

    await page.waitForChanges();

    validationText = await page.find(
      "ic-text-field >>> #ic-text-field-input-0-validation-text"
    );
    expect(validationText).toBeNull();

    element.setProperty("value", "0");

    await page.waitForChanges();

    validationText = await page.find(
      "ic-text-field >>> #ic-text-field-input-0-validation-text"
    );
    expect(validationText).toEqualText("Minimum value of 1 not met");
  });

  it("should focus input when setFocus method called", async () => {
    const page = await newE2EPage();
    await page.setContent('<ic-text-field label="Test label">');
    await page.waitForChanges();

    const element = await page.find("ic-text-field");
    await element.callMethod("setFocus");

    await page.waitForChanges();
    const activeElId = await page.$eval(
      "ic-text-field",
      (el) => el.shadowRoot.activeElement.id
    );

    expect(activeElId).toBe("ic-text-field-input-0");
  });

  it("works on a form", async () => {
    const page = await newE2EPage();

    await page.setContent(
      '<form><ic-text-field label="Test label" name="formInputEl"></ic-text-field><button id="resetButton" type="reset">Reset</button></form>'
    );
    await page.waitForChanges();

    let input = await page.find("ic-text-field >>> #ic-text-field-input-0");

    await input.press("s");
    await input.press("o");
    await input.press("m");
    await input.press("e");
    await input.press("Space");
    await input.press("t");
    await input.press("e");
    await input.press("x");
    await input.press("t");

    let value = await input.getProperty("value");
    expect(value).toBe("some text");
    //check value on underlying hidden form control
    const formInput = await page.find("input[name=formInputEl]");

    const visible = await formInput.isVisible();
    expect(visible).toBe(false);

    const formInputvalue = await formInput.getProperty("value");
    expect(formInputvalue).toBe("some text");

    const resetButton = await page.find("#resetButton");
    await resetButton.click();

    input = await page.find("ic-text-field >>> #ic-text-field-input-0");
    value = await input.getProperty("value");
    expect(value).toBe("");
  });
});
