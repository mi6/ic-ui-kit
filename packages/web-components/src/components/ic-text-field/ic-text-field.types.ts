export type IcTextFieldTypes =
  | "email"
  | "password"
  | "tel"
  | "text"
  | "url"
  | "number"
  | "search";

export type IcTextFieldInputModes =
  | "none"
  | "text"
  | "tel"
  | "url"
  | "email"
  | "numeric"
  | "decimal"
  | "search";

export type IcAriaAutocompleteTypes = "inline" | "list" | undefined;

export interface IcTextFieldKeydownEventDetail {
  event: KeyboardEvent;
  cursorPosition: number;
  selectionEnd: number;
}
