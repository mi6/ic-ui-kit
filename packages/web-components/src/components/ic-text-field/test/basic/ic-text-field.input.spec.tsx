import { TextField } from "../../ic-text-field";
import { newSpecPage } from "@stencil/core/testing";
import { waitForTimeout } from "../../../../testspec.setup";

describe("ic-text-field", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" rows=1></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders");
  });

  it("should render with placeholder", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" placeholder="placeholder" rows=0></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-placeholder");
  });

  it("should render with value", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-value");
  });

  it("should render with helperText, required and small", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" helper-text="helper text value" required=true small=true></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-helpertext");
  });

  it("should render with autoprops", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value" autofocus=true autocapitalize="on" autocomplete="on" autocorrect="on"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-autoprops");
  });

  it("should render with hidden label", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value" hide-label=true></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-hiden-label");
  });

  it("should render with icon", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label">
        <svg
          slot="icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0z" fill="none" />
          <path
            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-icon");
  });

  it("should render disabled", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value" disabled=true></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-disabled");
  });

  it("should render readonly", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value" readonly=true></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-readonly");
  });

  it("should render with max length", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="a long test value to exceed limit" max-length=25></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-maxlength");
  });

  it("should render with name & full width", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" value="test value" name="mycontolname" full-width=true></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-name-full-width");
  });

  it("should render with success validation", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" validation-status="success" validation-text="Good choice!" value="test value"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-success-validation");
  });

  it("should render with warning validation", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" validation-status="warning" validation-text="warning text" value="test value"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-warning-validation");
  });

  it("should render with error validation", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" validation-status="error" validation-text="error text" value="test value"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-error-validation");
  });

  it("should render with inline success validation", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" validation-status="success" validation-text="Good choice!" validation-inline="true" value="test value"></ic-text-field>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-inline-success-validation");
  });

  it("should focus", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" rows=1></ic-text-field>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icFocus", callbackFn);
    const input = page.root.shadowRoot.querySelector("input");
    await input.focus();
    expect(callbackFn).toHaveBeenCalled();
  });

  it("should blur", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" rows=1></ic-text-field>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icBlur", callbackFn);
    const input = page.root.shadowRoot.querySelector("input");
    await input.blur();
    expect(callbackFn).toHaveBeenCalled();
  });

  it("should not show left icon if text field is disabled", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" rows=1 disabled>
      <svg
          slot="icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0z" fill="none" />
          <path
            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </ic-text-field>`,
    });

    const slot = page.root.shadowRoot.querySelector(`[slot="icon"]`);
    expect(slot).toBeNull();
    expect(page.rootInstance.showLeftIcon).toBe(false);
  });

  it("should not exceed max length if new value < max length", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label" rows=1 max-length=5></ic-text-field>`,
    });

    page.rootInstance.watchValueHandler("test");
    expect(page.rootInstance.maxLengthExceeded).toBe(false);
  });

  it("should test keydown", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label"></ic-text-field>`,
    });

    const eventSpy = jest.fn();
    page.win.addEventListener("icKeydown", eventSpy);
    const event = new KeyboardEvent("keydown", { key: "A" });
    await page.rootInstance.handleKeyDown(event);
    await page.waitForChanges();

    expect(eventSpy).toHaveBeenCalled();
  });

  it("should test input", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<ic-text-field label="Test label"></ic-text-field>`,
    });

    const event = jest.fn();
    page.win.addEventListener("icInput", event);

    await page.rootInstance.onInput({ target: { value: "test value" } });
    await page.waitForChanges();
    expect(page.rootInstance.value).toBe("test value");
    expect(event).toHaveBeenCalled();
  });

  it("should test form reset event", async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<form>
        <ic-text-field label="Test label"></ic-text-field>
        <button id="resetButton" type="reset">Reset</button>  
      </form>`,
    });

    expect(page.rootInstance.value).toBe("");
    page.rootInstance.value = "New value";
    await page.waitForChanges();

    expect(page.rootInstance.value).toBe("New value");

    await page.rootInstance.handleFormReset();
    await page.waitForChanges();

    expect(page.rootInstance.value).toBe("");

    //test disconnected callback
    //delay to wait for icchange event to prevent console warning
    await waitForTimeout(500);
    page.setContent("");
  });
});

it("should render with min/max and max validation", async () => {
  const page = await newSpecPage({
    components: [TextField],
    html: `<ic-text-field label="Test label" rows=1 min=1 max=4 value="6" inputmode="numeric" type="number"></ic-text-field>`,
  });

  expect(page.root).toMatchSnapshot("renders-with-max");
});

it("should render with min/max and min validation", async () => {
  const page = await newSpecPage({
    components: [TextField],
    html: `<ic-text-field label="Test label" rows=1 min=1 max=4 value="0" inputmode="numeric" type="number"></ic-text-field>`,
  });

  expect(page.root).toMatchSnapshot("renders-with-min");
});
