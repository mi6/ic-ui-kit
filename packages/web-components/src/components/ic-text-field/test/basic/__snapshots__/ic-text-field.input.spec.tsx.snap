// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-text-field should render disabled: renders-disabled 1`] = `
<ic-text-field disabled="true" label="Test label" value="test value">
  <mock:shadow-root>
    <ic-input-container disabled="">
      <ic-input-label disabled="" for="ic-text-field-input-7" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container disabled="" validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" disabled="" id="ic-text-field-input-7" inputmode="text" name="ic-text-field-input-7" placeholder="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" disabled="" name="ic-text-field-input-7" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render readonly: renders-readonly 1`] = `
<ic-text-field label="Test label" readonly="" value="test value">
  <mock:shadow-root>
    <ic-input-container disabled="" readonly="">
      <ic-input-label for="ic-text-field-input-8" helpertext="" label="Test label" readonly=""></ic-input-label>
      <ic-input-component-container disabled="" readonly="" validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" class="no-left-pad readonly" disabled="" id="ic-text-field-input-8" inputmode="text" name="ic-text-field-input-8" placeholder="" readonly="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" disabled="" name="ic-text-field-input-8" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with autoprops: renders-with-autoprops 1`] = `
<ic-text-field autocapitalize="on" autocomplete="on" autocorrect="on" autofocus="true" label="Test label" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-4" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="on" autocomplete="on" id="ic-text-field-input-4" inputmode="text" name="ic-text-field-input-4" placeholder="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-4" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with error validation: renders-with-error-validation 1`] = `
<ic-text-field label="Test label" validation-status="error" validation-text="error text" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-13" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="error">
        <input aria-describedby="ic-text-field-input-13-validation-text" aria-invalid="true" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-13" inputmode="text" name="ic-text-field-input-13" placeholder="" type="text" value="test value">
      </ic-input-component-container>
      <ic-input-validation arialivemode="assertive" for="ic-text-field-input-13" message="error text" status="error"></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-13" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with helperText, required and small: renders-with-helpertext 1`] = `
<ic-text-field helper-text="helper text value" label="Test label" required="true" small="" value="">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-3" helpertext="helper text value" label="Test label" required=""></ic-input-label>
      <ic-input-component-container small="" validationstatus="">
        <input aria-describedby="ic-text-field-input-3-helper-text" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-3" inputmode="text" name="ic-text-field-input-3" placeholder="" required="" type="text" value="">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-3" type="hidden" value="">
</ic-text-field>
`;

exports[`ic-text-field should render with hidden label: renders-with-hiden-label 1`] = `
<ic-text-field hide-label="true" label="Test label" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-component-container validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-5" inputmode="text" name="ic-text-field-input-5" placeholder="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-5" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with icon: renders-with-icon 1`] = `
<ic-text-field label="Test label" value="">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-6" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="">
        <span slot="left-icon">
          <slot name="icon"></slot>
        </span>
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-6" inputmode="text" name="ic-text-field-input-6" placeholder="" type="text" value="">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0z" fill="none"></path>
    <path d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"></path>
  </svg>
  <input class="ic-input" name="ic-text-field-input-6" type="hidden" value="">
</ic-text-field>
`;

exports[`ic-text-field should render with inline success validation: renders-with-inline-success-validation 1`] = `
<ic-text-field label="Test label" validation-inline="true" validation-status="success" validation-text="Good choice!" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-14" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationinline="" validationstatus="success">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-14" inputmode="text" name="ic-text-field-input-14" placeholder="" type="text" value="test value">
      </ic-input-component-container>
      <ic-input-validation arialivemode="polite" for="ic-text-field-input-14" message="" status=""></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-14" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with max length: renders-with-maxlength 1`] = `
<ic-text-field label="Test label" max-length="25" value="a long test value to exceed limit">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-9" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="error">
        <input aria-describedby="ic-text-field-input-9-charcount-desc ic-text-field-input-9-validation-text" aria-invalid="true" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-9" inputmode="text" name="ic-text-field-input-9" placeholder="" type="text" value="a long test value to exceed limit">
      </ic-input-component-container>
      <ic-input-validation arialivemode="assertive" for="ic-text-field-input-9" message="Maximum length exceeded" status="error">
        <div slot="validation-message-adornment">
          <ic-typography class="error maxlengthtext" variant="caption">
            <span aria-live="polite" class="charcount" id="ic-text-field-input-9-charcount">
              33/25
            </span>
            <span hidden="" id="ic-text-field-input-9-charcount-desc">
              Field can contain a maximum of 25 characters.
            </span>
          </ic-typography>
        </div>
      </ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-9" type="hidden" value="a long test value to exceed limit">
</ic-text-field>
`;

exports[`ic-text-field should render with name & full width: renders-with-name-full-width 1`] = `
<ic-text-field class="fullwidth" full-width="true" label="Test label" name="mycontolname" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-10" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container fullwidth="" validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-10" inputmode="text" name="mycontolname" placeholder="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="mycontolname" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with placeholder: renders-with-placeholder 1`] = `
<ic-text-field label="Test label" placeholder="placeholder" rows="0" value="">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-1" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-1" inputmode="text" name="ic-text-field-input-1" placeholder="placeholder" type="text" value="">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-1" type="hidden" value="">
</ic-text-field>
`;

exports[`ic-text-field should render with success validation: renders-with-success-validation 1`] = `
<ic-text-field label="Test label" validation-status="success" validation-text="Good choice!" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-11" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="success">
        <input aria-describedby="ic-text-field-input-11-validation-text" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-11" inputmode="text" name="ic-text-field-input-11" placeholder="" type="text" value="test value">
      </ic-input-component-container>
      <ic-input-validation arialivemode="polite" for="ic-text-field-input-11" message="Good choice!" status="success"></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-11" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with value: renders-with-value 1`] = `
<ic-text-field label="Test label" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-2" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-2" inputmode="text" name="ic-text-field-input-2" placeholder="" type="text" value="test value">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-2" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render with warning validation: renders-with-warning-validation 1`] = `
<ic-text-field label="Test label" validation-status="warning" validation-text="warning text" value="test value">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-12" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="warning">
        <input aria-describedby="ic-text-field-input-12-validation-text" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-12" inputmode="text" name="ic-text-field-input-12" placeholder="" type="text" value="test value">
      </ic-input-component-container>
      <ic-input-validation arialivemode="polite" for="ic-text-field-input-12" message="warning text" status="warning"></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-12" type="hidden" value="test value">
</ic-text-field>
`;

exports[`ic-text-field should render: renders 1`] = `
<ic-text-field label="Test label" rows="1" value="">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-0" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="">
        <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-0" inputmode="text" name="ic-text-field-input-0" placeholder="" type="text" value="">
      </ic-input-component-container>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-0" type="hidden" value="">
</ic-text-field>
`;

exports[`should render with min/max and max validation: renders-with-max 1`] = `
<ic-text-field inputmode="numeric" label="Test label" max="4" min="1" rows="1" type="number" value="6">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-22" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="error">
        <input aria-describedby="ic-text-field-input-22-validation-text" aria-invalid="true" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-22" inputmode="numeric" max="4" min="1" name="ic-text-field-input-22" placeholder="" type="number" value="6">
      </ic-input-component-container>
      <ic-input-validation arialivemode="assertive" for="ic-text-field-input-22" message="Maximum value of 4 exceeded" status="error"></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-22" type="hidden" value="6">
</ic-text-field>
`;

exports[`should render with min/max and min validation: renders-with-min 1`] = `
<ic-text-field inputmode="numeric" label="Test label" max="4" min="1" rows="1" type="number" value="0">
  <mock:shadow-root>
    <ic-input-container>
      <ic-input-label for="ic-text-field-input-23" helpertext="" label="Test label"></ic-input-label>
      <ic-input-component-container validationstatus="error">
        <input aria-describedby="ic-text-field-input-23-validation-text" aria-invalid="true" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-23" inputmode="numeric" max="4" min="1" name="ic-text-field-input-23" placeholder="" type="number" value="0">
      </ic-input-component-container>
      <ic-input-validation arialivemode="assertive" for="ic-text-field-input-23" message="Minimum value of 1 not met" status="error"></ic-input-validation>
    </ic-input-container>
  </mock:shadow-root>
  <input class="ic-input" name="ic-text-field-input-23" type="hidden" value="0">
</ic-text-field>
`;
